src/utils.jl

```diff
@@ -96,6 +96,101 @@ function classify(path::OrbitPath, axis)
     return classify(path.r,path.z,path.pitch, axis, n=length(path))
 end
 
+"""
+
+gcde_check(M, o; verbose=false)
+
+For each point along the orbit path, compute the gyro-radius r_g, the magnetic field
+amplitude B and the square root of the maximum eigenvalue of the matrix ̂M. Use these quantities
+to determine whether the usage of guiding-center drift equations (GCDE) could be expected to produce 
+a significant discrepancy compared to full-orbit equations (FOE). The matrix ̂M is the matrix
+found in equation (2) of D. Pfefferlé et al (2015) (https://doi.org/10.1088/0741-3335/57/5/054017)
+and the condition to be evaluated is the condition found in equation (3) of the same paper.
+In section 3 of the paper, the abritrary threshold of 0.073 is used. The default is the same here.
+If the condition is satisfied for the threshold, no significant discrepancy between GCDE 
+and FOE is to be expected. The equations to compute ̂M are from Appendix B of the same paper (B.2).
+
+Return true if the criterion is fulfilled, and it is ok to use GCDE.
+Return false if the criterion is not fulfilled, and FOE should be used instead.
+"""
+function gcde_check(M::AbstractEquilibrium, o::Orbit; threshold=0.073, verbose=false, nr = 33, nz = 33, kappa=1.4)
+
+    (typeof(M) <: SolovevEquilibrium) && (kappa=M.kappa)
+    Rmax = maximum(o.path.r)
+    Rmin = minimum(o.path.r)
+    dR = (Rmax-Rmin)/nr
+    r = range(Rmin-dR,stop=Rmax+dR,length=nr) # A reasonable r range
+    z = kappa*range(Rmin-dR,stop=Rmax+dR,length=nr) # A reasonable z range
+    g_rz = zeros(length(r),length(z))
+    for (ri,rr)=enumerate(r),(zi,zz)=enumerate(z)
+        g_rz[ri,zi] = poloidal_current(M,M(rr,zz))
+    end
+    g_rz_itp = Interpolations.CubicSplineInterpolation((r,z), g_rz, extrapolation_bc=Interpolations.Flat()) # Poloidal current function (F=R*Bt) as a function of R,Z
```

Comment:
Any reason you made an interpolation instead of just using an anonymous function `g_rz = (r,z) -> poloidal_current(M, M(r,z))` which you can then pass to ForwardDiff

Reply:
No reason at all. Updated the code now. Thanks